service/controller/app/v1/key/key.go

```diff
@@ -153,3 +159,15 @@ func VersionLabel(customResource v1alpha1.App) string {
 		return ""
 	}
 }
+
+func GenerateTarballURL(baseURL string, appName string, version string) (string, error) {
```

Comment:
IMO we should stop adding key functions that return errors. When the implementation is so complex it may not be a key function.

Reply:
That makes sense but we need to reuse this logic between the chart and chartoperator resources. What about defining a tarball package for this?

Reply:
`pkg/tarball` sounds nice.